collectors/mimecast/utils.js

```diff
@@ -40,11 +40,14 @@ function getAPILogs(authDetails, state, accumulator, maxPagesPerInvocation) {
                     pageCount++;
                     switch (state.stream) {
                         case Siem_Logs:
-                            if (body.data) {
+                            if (body.data && body.data[0]) {
                                 accumulator.push(...body.data);
                             }
                             if (response.meta && response.meta.isLastToken) {
                                 nextPage = undefined;
+                                if(applicationDetails.payload.data[0].token){
```

Comment:
same here. Is it possible `applicationDetails.payload.data` is an empty array?

Reply:
here, it's not possible because we are generating payload while run API